### PR TITLE
feat: Add image pull secrect support for python src

### PR DESCRIPTION
**Pull request by lcgash**
What changes were proposed in this pull request?

add image pull secret config for workflow

Why are the changes needed?

because our project need pull image from private hub

Does this PR introduce any user-facing change?

i think no.it was a new feature.

How was this patch tested?

add one unit test and our project use this code to pull image

### CODE REVIEW

File: couler/core/syntax/__init__.py [hunk 1]
```diff
@@ -22,3 +22,6 @@
     add_volume,
     create_workflow_volume,
 )
+from couler.core.syntax.image_pull_secret import (  # noqa: F401
+    add_image_pull_secret,
+)
```
Comment (merlintang) [inline couler/core/syntax/__init__.py:27]: please update this

**Comment by merlintang** [comment]
@lcgash thanks for your contribution, can you fix the type issue at first?

File: couler/core/syntax/__init__.py (hunk 1 above)
Comment (lcgash) [reply couler/core/syntax/__init__.py:27]: sorry, this is my first time submitting pr in github, how should i fix it?

Comment (lcgash) [reply couler/core/syntax/__init__.py]: look like done. please check, thanks

**Comment by merlintang** [comment]
@lcgash can you update your pr description as well? 

What changes were proposed in this pull request?
Why are the changes needed?
Does this PR introduce any user-facing change?
How was this patch tested?

**Comment by merlintang** [comment]
https://yuque.antfin.com/aii/kubemaker/mz5myx @lcgash  pls run precommit to help you to fix the format issue

**Comment by lcgash** [comment]
> https://yuque.antfin.com/aii/kubemaker/mz5myx @lcgash pls run precommit to help you to fix the format issue
sorry , i can not access it

**Comment by lcgash** [comment]
> pre-commit run --all
> @lcgash pls run precommit to help you to fix the format issue. sorry about this link error https://github.com/couler-proj/couler/blob/master/CONTRIBUTING.md#run-sanity-checks

i got

**Comment by lcgash** [comment]
> pre-commit run --all
> @lcgash pls run precommit to help you to fix the format issue. sorry about this link error https://github.com/couler-proj/couler/blob/master/CONTRIBUTING.md#run-sanity-checks

image
it seems look ok. please review, thanks

**Review by merlintang — APPROVED** [review]

**Comment by merlintang** [comment]
thanks @lcgash  for your contribution. can you also send a pr to add your organization to the adopter of couler. https://github.com/couler-proj/couler/blob/master/ADOPTERS.md

**Pull request merged**